File: mealpository-frontend/src/app/models/recipe.ts
import { Ingredient } from "./ingredient";
import { Instruction } from "./instruction";

export class Recipe {
    
    recipe_id: string;
    title: string;
    description: string;
    servings: string;
    prep_time: string;
    cook_time: string;
    total_time: string;
    image_url: string;
    user_id: string;
    ingredients: Ingredient[];
    instructions: Instruction[];

    constructor (
        recipe_id: string,
        title: string,
        description: string,
        servings: string,
        prep_time: string,
        cook_time: string,
        total_time: string,
        image_url: string,
        user_id: string,
        ingredients: Ingredient[],
        instructions: Instruction[]
    ) {
        this.recipe_id = recipe_id;
        this.title = title;
        this.description = description;
        this.servings = servings;
        this.prep_time = prep_time;
        this.cook_time = cook_time;
        this.total_time = total_time;
        this.image_url = image_url;
        this.user_id = user_id;
        this.instructions = instructions;
        this.ingredients = ingredients;
    }

}

File: mealpository_backend/database/recipe.py
class Recipe:
    def __init__(self, 
                 recipe_id, title, description, servings, prep_time,
                   cook_time, total_time, image_url, user_id, file_name, ingredients, instructions
                ):
        self.recipe_id = recipe_id
        self.title = title
        self.description = description
        self.servings = servings
        self.prep_time = prep_time
        self.cook_time = cook_time
        self.total_time = total_time
        self.image_url = image_url
        self.user_id = user_id
        self.file_name = file_name
        self.ingredients = ingredients
        self.instructions = instructions

    def serialize_self(self):
        return {
            'recipe_id': self.recipe_id,
            'title': self.title,
            'description': self.description,
            'servings': self.servings,
            'prep_time': self.prep_time,
            'cook_time': self.cook_time,
            'total_time': self.total_time,
            'image_url': self.image_url,
            'user_id': self.user_id,
            'file_name': self.file_name,
            'ingredients': self.ingredients,
            'instructions': self.instructions
        }
